n3fit/src/n3fit/model_gen.py

```diff
@@ -535,6 +535,7 @@ def pdfNN_layer_generator(
 
     # First prepare the input for the PDF model and any scaling if needed
     placeholder_input = Input(shape=(None, 1), batch_size=1)
+    # placeholder_ph = ???
```

Comment:
No, the placeholder input will be feed to the NN, is the same for everyone.

You will need to store at some point the union of all values of x for the photon though.

Reply:
Indeed, this is were all the problem came from: 
- it is not possible to store the values for the photon in a layer, since layers are defined in here, but x values are not known yet, so the photon can't be evaluated
- but the photon is constant, so you do not want to evaluate more than once

My, possibly poor solution, was to pass it as an input to the NN, and propagate the input to the output through a layer.
Of course, even better would be to directly hijack the values on the output to a set of constants, but they are constants once the x values are known.

If you know a way to do it without passing them as input to the NN I'd be interested in it. Never played myself with `n3fit`, so I rely on the expert :)